client/state/data-layer/wpcom/sites/plugins/recommended/index.js

```diff
@@ -0,0 +1,53 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+/**
+ * Internal dependencies
+ */
+import { normalizePluginsList } from 'lib/plugins/utils';
+import { RECOMMENDED_PLUGINS_REQUEST } from 'state/action-types';
+import { receiveRecommendedPlugins } from 'state/plugins/recommended/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+
+export const fetch = action => {
+	const { siteId, limit = 6 } = action;
+
+	return [
+		recordTracksEvent( 'calypso_recommended_plugins_requested', { site_id: siteId, limit } ),
```

Comment:
Could we verify we're planning to use these events for something? I assume that's the case, but it feels very generic to be tracking when plugins are requested and received - we usually track events that are more tightly coupled with what the user is doing or seeing, and not with what the app is doing internally.

Reply:
We're planning on using these events to determine relability and quality of our recommendations service; I know it's not exactly "user behavior" tracking which we'd normally use Tracks for, but I wasn't sure if there was something else we could use for this purpose.

Do you have any recommendations?

Reply:
I have no hard feelings on usage of tracks events for such needs, and to be honest, it won't be a precedent. 

But the purposes you explain seem to demonstrate that it might be better to track that information on the server side instead. 

Understandably, we have better tools for those kinds of logging on the server side, for example I'd recommend using `bump_stats_extras()` and `log2logstash()` (depending on the level of the detail of data you want to store). 

Let me know what you think!

Reply:
After giving this a bit of thought, I agree that it'd better to this kind of logging on the server-side. 👍